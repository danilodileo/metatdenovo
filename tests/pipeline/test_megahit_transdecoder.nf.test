nextflow_pipeline {
    name "Test pipeline"
    script "../../main.nf"
    tag "test_megahit_transdecoder"
    tag "PIPELINE"

    test("Test MEGAHIT/transdecoder") {

        when {
            params {
                outdir = "$outputDir"
                input = 'https://raw.githubusercontent.com/nf-core/test-datasets/metatdenovo/samplesheet/samplesheet.csv'
                assembler      = 'megahit'
                orf_caller     = 'transdecoder'
                skip_eukulele  = true
                skip_eggnog    = true
                skip_kofamscan = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/trimgalore/SAMPLE1_PE_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/fastqc/SAMPLE1_PE_1_val_1_fastqc.html").exists() },
                { assert new File("$outputDir/megahit/megahit_out/megahit_assembly.log").exists() },
                { assert new File("$outputDir/samtools/SAMPLE1_PE.sorted.bam.bai").exists() },
                { assert new File("$outputDir/bbmap/logs/SAMPLE1_PE.bbmap.log").exists() },
                { assert new File("$outputDir/featurecounts/SAMPLE1_PE.featureCounts.tsv.summary").exists() },
                { assert new File("$outputDir/transdecoder/megahit.transdecoder/megahit_assembly.contigs.fa.transdecoder.bed").exists() },
                { assert new File("$outputDir/transdecoder/megahit.transdecoder/megahit_assembly.contigs.fa.transdecoder.cds").exists() },
                { assert new File("$outputDir/transdecoder/megahit.transdecoder/megahit_assembly.contigs.fa.transdecoder.gff3").exists() },
                { assert new File("$outputDir/transdecoder/megahit.transdecoder/megahit_assembly.contigs.fa.transdecoder.pep").exists() },
                { assert new File("$outputDir/summary_tables/megahit.transdecoder.counts.tsv.gz").exists() },
                { assert new File("$outputDir/summary_tables/megahit.transdecoder.overall_stats.tsv.gz").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert new File("$outputDir/pipeline_info/nf_core_metatdenovo_software_mqc_versions.yml").exists() }
            )
        }
    }
}
