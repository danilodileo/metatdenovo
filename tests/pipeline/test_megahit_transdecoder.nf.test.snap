{
    "SAMPLE1_PE_1_val_1_fastqc.html": {
        "content": [
            "SAMPLE1_PE_1_val_1_fastqc.html:md5,89c71d5bf2499665b8c036a2ffae3bd7"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T18:03:34.857244"
    },
    "SAMPLE1_PE.bbmap.log": {
        "content": [
            "SAMPLE1_PE.bbmap.log:md5,2a19e45249e0557291e2b0c06df53d10"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T20:35:33.833897"
    },
    "megahit_assembly.log": {
        "content": [
            "megahit_assembly.log:md5,6cc1f0dd89486040c4ace58d46235db3"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T20:35:33.823598"
    },
    "multiqc_report.html": {
        "content": [
            "multiqc_report.html:md5,dfbd537831ab64fcfa8cc5c1f5094356"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T20:35:33.893689"
    },
    "SAMPLE1_PE.featureCounts.tsv.summary": {
        "content": [
            "SAMPLE1_PE.featureCounts.tsv.summary:md5,a0cd2da6a20224af6c7693a4cfbdd2dc"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T20:35:33.847641"
    }
}